From 81b35b39cb94b502b91ea5903e3bd5ec6368755a Mon Sep 17 00:00:00 2001
From: Jason Rogena <j.rogena@cgiar.org>
Date: Tue, 17 Jun 2014 15:42:20 +0300
Subject: [PATCH 203/213] Version 1.0.4: Edit first cow bug

Fixed bug where admins could only edit the first cow.
---
 MistroFarmer/src/main/AndroidManifest.xml                      |  4 ++--
 .../org/cgiar/ilri/mistro/farmer/CowSelectionActivity.java     | 10 ++++++----
 .../java/org/cgiar/ilri/mistro/farmer/EditCowActivity.java     |  3 +++
 .../java/org/cgiar/ilri/mistro/farmer/backend/DataHandler.java |  2 +-
 .../main/java/org/cgiar/ilri/mistro/farmer/carrier/Cow.java    |  4 ++++
 .../main/java/org/cgiar/ilri/mistro/farmer/carrier/Farmer.java |  2 +-
 6 files changed, 17 insertions(+), 8 deletions(-)

diff --git a/MistroFarmer/src/main/AndroidManifest.xml b/MistroFarmer/src/main/AndroidManifest.xml
index 6caaf0e..966fd13 100644
--- a/MistroFarmer/src/main/AndroidManifest.xml
+++ b/MistroFarmer/src/main/AndroidManifest.xml
@@ -1,8 +1,8 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="org.cgiar.ilri.mistro.farmer"
-    android:versionCode="18"
-    android:versionName="1.0.3" >
+    android:versionCode="19"
+    android:versionName="1.0.4" >
 
     <uses-sdk
         android:minSdkVersion="7"
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/CowSelectionActivity.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/CowSelectionActivity.java
index 5174d11..5ff8d5c 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/CowSelectionActivity.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/CowSelectionActivity.java
@@ -204,7 +204,7 @@ public class CowSelectionActivity extends SherlockActivity implements MistroActi
                         newlyFilteredFarmers.add(currFarmer);
                     }
                     else{
-                        Log.d(TAG, currFarmer.getExtensionPersonnel()+" does not qualify to be null");
+                        //Log.d(TAG, currFarmer.getExtensionPersonnel()+" does not qualify to be null");
                     }
                 }
 
@@ -221,7 +221,7 @@ public class CowSelectionActivity extends SherlockActivity implements MistroActi
                         newlyFilteredFarmers.add(currFarmer);
                     }
                     else{
-                        Log.d(TAG, currFarmer.getExtensionPersonnel()+" does not match "+selection);
+                        //Log.d(TAG, currFarmer.getExtensionPersonnel()+" does not match "+selection);
                     }
                 }
 
@@ -286,10 +286,12 @@ public class CowSelectionActivity extends SherlockActivity implements MistroActi
                         if(cows != null){
                             Cow selectedCow = cows.get(selectCowS.getSelectedItemPosition());
                             if(selectedCow != null){
+                                Log.d(TAG, "Cow index = "+selectCowS.getSelectedItemPosition());
+                                Log.d(TAG, "Number of cows = "+cows.size());
                                 Intent intent=new Intent(this, EditCowActivity.class);
                                 intent.putExtra(MainMenu.KEY_ADMIN_DATA, adminData.toString());
-                                intent.putExtra(CowRegistrationActivity.KEY_INDEX, selectCowS.getSelectedItemPosition());
-                                intent.putExtra(CowRegistrationActivity.KEY_NUMBER_OF_COWS, cows.size());
+                                intent.putExtra(EditCowActivity.KEY_INDEX, selectCowS.getSelectedItemPosition());
+                                intent.putExtra(EditCowActivity.KEY_NUMBER_OF_COWS, cows.size());
                                 Bundle bundle=new Bundle();
                                 bundle.putParcelable(Farmer.PARCELABLE_KEY, selectedFarmer);
                                 intent.putExtras(bundle);
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/EditCowActivity.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/EditCowActivity.java
index b0e9ef9..c4de847 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/EditCowActivity.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/EditCowActivity.java
@@ -317,6 +317,9 @@ public class EditCowActivity extends SherlockActivity implements MistroActivity,
             if(farmer!=null){
                 thisCow = farmer.getCow(index);
                 if(thisCow != null) {
+                    Log.d(TAG, "ID for current cow is "+String.valueOf(thisCow.getId()));
+                    Log.d(TAG, "Index from bundle = "+String.valueOf(index));
+                    Log.d(TAG, "Number of animals from bundle = "+String.valueOf(numberOfCows));
                     nameET.setText(thisCow.getName());
                     earTagNumberET.setText(thisCow.getEarTagNumber());
                     dateOfBirthET.setText(thisCow.getDateOfBirth());
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/DataHandler.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/DataHandler.java
index be939d3..eff8ee8 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/DataHandler.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/DataHandler.java
@@ -480,7 +480,7 @@ public class DataHandler
      */
     public static String getSharedPreference(Context context, String sharedPreferenceKey, String defaultValue) {
         SharedPreferences sharedPreferences = context.getSharedPreferences(context.getString(R.string.app_name), Context.MODE_PRIVATE);
-        Log.d(TAG, "value of " + sharedPreferenceKey + " is " + sharedPreferences.getString(sharedPreferenceKey, defaultValue));
+        //Log.d(TAG, "value of " + sharedPreferenceKey + " is " + sharedPreferences.getString(sharedPreferenceKey, defaultValue));
         return sharedPreferences.getString(sharedPreferenceKey, defaultValue);
     }
 
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/carrier/Cow.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/carrier/Cow.java
index f6d0683..6c537e2 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/carrier/Cow.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/carrier/Cow.java
@@ -149,6 +149,10 @@ public class Cow implements Parcelable, Serializable {
         }
     }
 
+    public int getId(){
+        return this.id;
+    }
+
     private void initFromJSON(JSONObject jsonObject){
         try {
             id = jsonObject.getInt("id");
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/carrier/Farmer.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/carrier/Farmer.java
index 10aa15e..c623dac 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/carrier/Farmer.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/carrier/Farmer.java
@@ -63,7 +63,7 @@ public class Farmer implements Parcelable, Serializable
 
     public Farmer(JSONObject farmerJsonObject){
         try{
-            Log.d(TAG, "Json for this farmer = "+farmerJsonObject.toString());
+            //Log.d(TAG, "Json for this farmer = "+farmerJsonObject.toString());
             id = farmerJsonObject.getInt("id");
             fullName = farmerJsonObject.getString("name");
             this.extensionPersonnel = (DataHandler.isNull(farmerJsonObject.getString("extension_personnel"))) ? "" : farmerJsonObject.getString("extension_personnel");
-- 
2.7.1

